src/views/preview/preview.jsx

```diff
@@ -67,6 +69,18 @@ class Preview extends React.Component {
         this.getExtensions(this.state.projectId);
         this.addEventListeners();
     }
+    componentWillUpdate (nextProps, nextState) {
+        if (this.props.reportStatus !== nextProps.reportStatus) {
+            this.setState({
+                report: {
+                    ...nextState.report,
+                    confirmed: (nextProps.reportStatus === previewActions.Status.FETCHED),
+                    error: (nextProps.reportStatus === previewActions.Status.ERROR),
+                    waiting: (nextProps.reportStatus === previewActions.Status.FETCHING)
```

Comment:
It's not recommended to set state based on props for the same reason that it's a linter error to `setState` in `componentDidUpdate` — it's possible to cause unnecessary (or infinite) re-renders. Since these are all calculations based on current props, then this is probably best placed in `render` where we pass this state on to the next component.

Since this is coming out of redux, maybe the modal should just get it from there directly?

Reply:
Done. See new code for modal/report/modal.jsx